dvc/scm/git/backend/dulwich.py

```diff
@@ -101,8 +101,6 @@ def add(self, paths: Iterable[str]):
 
         files = []
         for path in paths:
-            if not os.path.isabs(path):
-                path = os.path.join(self.root_dir, path)
```

Comment:
`path` will be relative from cwd which may not be SCM root (which is already handled properly in the line 113 `relpath` call)

Reply:
If we're in a submodule, we do need to do this adjustment (but not for any other relative path).